crates/flowctl/src/raw/suggest_schema.rs

```diff
@@ -0,0 +1,222 @@
+use anyhow::Context;
+use bytes::Bytes;
+use doc::{inference::Shape, SchemaIndexBuilder, FailedValidation};
+use futures::{TryStreamExt, StreamExt};
+use journal_client::{broker, read::uncommitted::{ReadUntil, ReadStart, JournalRead, ExponentialBackoff, Reader}, list::list_journals, AuthHeader};
+use json::schema::{build::build_schema, types};
+use models::{
+    Capture, CaptureBinding, CaptureDef, CaptureEndpoint, Catalog, Collection, CollectionDef,
+    CompositeKey, ConnectorConfig, JsonPointer, Schema, ShardTemplate,
+};
+use proto_flow::{
+    capture::{request, Request},
+    flow::capture_spec::ConnectorType,
+    ops::Log
+};
+use bytelines::AsyncByteLines;
+use proto_grpc::broker::journal_client::JournalClient;
+use schema_inference::{json_decoder::JsonCodec, inference::infer_shape, shape, server::InferenceError, schema::SchemaBuilder};
+use serde_json::{json, value::RawValue, Value};
+use tokio::io::{BufReader, AsyncReadExt};
+use tokio_util::{compat::FuturesAsyncReadCompatExt, io::{ReaderStream, StreamReader}, codec::FramedRead};
+use tonic::{codegen::InterceptedService, transport::Channel};
+use std::{collections::BTreeMap, io::ErrorKind, time::{Duration, Instant}};
+use url::Url;
+
+use crate::{connector::docker_run, catalog::{fetch_live_specs, List, SpecTypeSelector, NameSelector, collect_specs}, dataplane::{self, fetch_data_plane_access_token}};
+use crate::local_specs;
+use anyhow::anyhow;
+
+#[derive(Debug, clap::Args)]
+#[clap(rename_all = "kebab-case")]
+pub struct SuggestSchema {
+    /// Collection name
+    #[clap(long)]
+    collection: String,
+
+    /// Task name
+    #[clap(long)]
+    task: String,
+}
+
+pub async fn do_suggest_schema(
+    ctx: &mut crate::CliContext,
+    SuggestSchema {
+        collection,
+        task,
+    }: &SuggestSchema,
+) -> anyhow::Result<()> {
+    let client = ctx.controlplane_client().await?;
+    // Retrieve identified live specifications.
+    let live_specs = fetch_live_specs(
+        client.clone(),
+        &List {
+            flows: true,
+            name_selector: NameSelector {
+                name: vec![collection.clone()],
+                prefix: Vec::new()
+            },
+            type_selector: SpecTypeSelector {
```

Comment:
@psFried the default for `SpecTypeSelector` fields is `None`, and `None` behaves the same as `Some(true)`, see [the comments there](https://github.com/estuary/flow/blob/master/crates/flowctl/src/catalog/mod.rs#L122-L127):

````
/// Whether to include captures in the selection
    ///
    /// If true, or if no value was given, then captures will
    /// be included. You can also use `--captures=false` to exclude
    /// captures.
```

Also see [this logic](https://github.com/estuary/flow/blob/master/crates/flowctl/src/catalog/mod.rs#L163C63-L170) which explicitly unwraps the `Option` value with a default value of `false`.

Reply:
Actually I'm confused about the second logic that I linked above, if the `Option` is unwrapped to `false` and the spec type is filtered out, does that mean we just don't specify any query for that spec type, and does that mean it will be included by default?

Reply:
The logic for this is [here](https://github.com/estuary/flow/blob/58b6586fa160340955c0ea6806b9676b8174d654/crates/flowctl/src/catalog/mod.rs#L175-L197). If any `SpecTypeSelector` fields are `Some(true)`, then it'll query for specs having only that spec type. If none are `true` and some are `Some(false)`, then it will only filter out the types that you don't want. If all fields are `None`, then it just returns everything.